### PR TITLE
Allow Irrlicht caching of animated meshes

### PR DESCRIPTION
**Pull request by lhofhansl**
Possible solution for #6676 !

Instead of setting the color of each vertex (which is shared between instances of the same animated mesh), set emissive color instead, which is unique per instance of the mesh.

That way we can use Irrlicht's mesh cache.

This is for testing. Please have a look at this and test it out.

### CODE REVIEW

**Comment by lhofhansl** [comment]
Note that when we go real lighting, this will have to be thought through again.

File: src/client.cpp [hunk 1]
```diff
@@ -1855,10 +1855,7 @@ scene::IAnimatedMesh* Client::getMesh(const std::string &filename)
 
 	scene::IAnimatedMesh *mesh = RenderingEngine::get_scene_manager()->getMesh(rfile);
 	rfile->drop();
-	// NOTE: By playing with Irrlicht refcounts, maybe we could cache a bunch
-	// of uniquely named instances and re-use them
 	mesh->grab();
-	RenderingEngine::get_mesh_cache()->removeMesh(mesh);
```
Comment (lhofhansl) [inline src/client.cpp]: Note that this can potentially be improved further. MT's mesh data cache is no longer needed.
But I'd like to leave this for another PR.

**Comment by lhofhansl** [comment]
I played around with this. Used torches and other light sources, day and night. This seems to work fine in all scenarios.
Even tried the torches mod that allows for a wielded torch, all the active objects were lit correctly.

I think we can remove the WIP label.

**Comment by Fixer-007** [comment]
In singleplayer, problems with rendering of gates (no rendering in world, broken icon):
default
In singleplayer, install homedecor mod and observe smth like this:
default
In multiplayer, join Time Travel Server to observe smth like this:
default

**Comment by lhofhansl** [comment]
Thanks @Fixer-007 will look into these.

Strange on the doors, and home decor items, I'm not changing any of the geometries.
How is the last scene supposed to look?

**Comment by lhofhansl** [comment]
How about this @Fixer-007 ?
Generally changing client::getMesh(...) was too radical. Changed is such that the default is not using the Irrlicht, and animated meshes can override this. This fixes the door issue in the inventory. Will also test home decor now.

**Comment by Fixer-007** [comment]
Can confirm, bug was fixed.

**Comment by lhofhansl** [comment]
Anything else I need to do here? :)

**Comment by sfan5** [comment]
Tested and works :+1:

**Comment by Fixer-007** [comment]
Found another bug, it seems, player skin is not rendered on jordach.net port 30000.

**Comment by lhofhansl** [comment]
Thanks @sfan5 

This should not affect skins. I will of course double check, before I merge.

Thanks for the thorough testing @Fixer-007 .

**Comment by lhofhansl** [comment]
@Fixer-007 Do you know which mod that is?

**Comment by lhofhansl** [comment]
I tried the 3d-armor mod and wardrobe mod. All works correctly.

I can only reproduce that with that server. :(

GenericCAO is also used for players, I think I should disable Irrlicht's meshcaching for players.
Will verify that theory now. :)

**Comment by lhofhansl** [comment]
OK... Found it. Not caching the player's mesh and initially setting vertex colors fixes the problem.
Not it's absolutely only done for active object meshes.

**Comment by lhofhansl** [comment]
Here's the change. No model should have vertexes with alpha set to 0 (transparent), but since it appears to be case here, this now making sure an alpha is set.

I'll merge this later tonight unless I see more comments.

**Comment by lhofhansl** [comment]
fd9f195f
